mcux/CMakeLists.txt

```diff
@@ -36,6 +37,9 @@ if (${MCUX_DEVICE} MATCHES "LPC")
   zephyr_sources(devices/${MCUX_DEVICE}/fsl_power.c)
   zephyr_sources(devices/${MCUX_DEVICE}/fsl_reset.c)
 endif()
+if (${MCUX_SERIES} MATCHES "K2X")
+  zephyr_sources(devices/${MCUX_DEVICE}/system_${MCUX_DEVICE}.c)
```

Comment:
This breaks the convention of having the setup code of the SOC in zephyr itself rather than in the HAL

Reply:
How 'established' is this convention?

E.g. nordic does this the same, see https://github.com/zephyrproject-rtos/hal_nordic/blob/master/CMakeLists.txt#L23

Reply:
I think the point is that all other NXP SoCs do it as @brog mentions and the K22F support should match what other NXP Kinetis SoCs are doing.